src/python/Utils/Utilities.py

```diff
@@ -67,7 +67,7 @@ def strToBool(string):
 
     Try to convert a string to boolean. i.e. "True" to python True
     """
-    if string in [False, True]:
+    if string is False or string is True:
```

Comment:
Hi @amaltaro,
It seems to me this line is not going to work. Maybe I do not get the logic here... but in any way it seems to be relying on the internal python interpretation of a `str` or `int` value to True or False. And since this input value may not be a string here but also an integer now, then it magically falls under the proper true/false category at this line. And this to me is a quite badly hidden logic. And sorry if I did it wrong, but since I had difficulties going in my mind though all possible cases/combinations of inputs to this function, such as `1`| `'1'`| `0` |`'0'`|`'True'`|`'False'` etc. I decided to test it in python and here is the result: 
```
(WMCore.venv3) user@host:WMCoreDev.d$ ipython 
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: def strToBool(string):
   ...:     """
   ...:     _strToBool_
   ...:     Try to convert a string to boolean. i.e. "True" to python True
   ...:     """
   ...:     if string is False or string is True:
   ...:         return string
   ...:     elif string in ["True", "true", "TRUE"]:
   ...:         return True
   ...:     elif string in ["False", "false", "FALSE"]:
   ...:         return False
   ...:     raise ValueError("Can't convert to bool: %s" % string)
   ...: 
```
No matter if I test it with a string or an integer ....same result:

```
In [2]: strToBool('1')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 strToBool('1')

Input In [1], in strToBool(string)
     10 elif string in ["False", "false", "FALSE"]:
     11     return False
---> 12 raise ValueError("Can't convert to bool: %s" % string)

ValueError: Can't convert to bool: 1

In [3]: strToBool(1)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Input In [3], in <cell line: 1>()
----> 1 strToBool(1)

Input In [1], in strToBool(string)
     10 elif string in ["False", "false", "FALSE"]:
     11     return False
---> 12 raise ValueError("Can't convert to bool: %s" % string)

ValueError: Can't convert to bool: 1

``` 

On the other hand I do see there is a more standardized way of doing it with an upstream function:

```
In [4]: from distutils.util import strtobool

In [5]: strtobool('1')
Out[5]: 1

In [6]: strtobool('0')
Out[6]: 0

In [7]: bool(strtobool('0'))
Out[7]: False

In [8]: bool(strtobool('1'))
Out[8]: True

In [9]: bool(strtobool('y'))
Out[9]: True

In [10]: bool(strtobool('true'))
Out[10]: True

In [11]: bool(strtobool('True'))
Out[11]: True

In [12]: bool(strtobool('False'))
Out[12]: False
``` 

here is the help to this function:
```
In [16]: help(strtobool)

strtobool(val)
    Convert a string representation of truth to true (1) or false (0).
    
    True values are 'y', 'yes', 't', 'true', 'on', and '1'; false values
    are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
    'val' is anything else.

```

Reply:
Hi Todor, I didn't know this library/function exist, however there are two problems with this:
* according to the documentation, it's planned to be deprecated in python 3.12
* and it would break compatibility with our code, since it does not consider "TRUE" or "True" to be `True`.

Regarding your tests, it has the same expected results as mine. In short:
* if input is an integer, then we do not map it to a boolean and validation will fail
* if it's a boolean, then validation is successful
* if it's a string matching the 6 string representations that we implemented for True and False, then it's successful
* if it's any other string, even "1", it will fail.

Does it make sense now?

Reply:
Yes, it does make sense. I did notice this logic right after I have sent the review.... which is a sign that it is kind of twisted/reversed and easy to get lost in. If you decide to keep it that way, could you please fix this docstring and mention this somewhere, so next time while reading it we (me in particular) do not need to scratch heads. 
And at least it would worth mentioning that "except a pure string the input value could also be a bool". Thanks!

Reply:
That's fair! Indeed a better docstring will make it better. I have just updated it. Thanks for the suggestion.